### DISCUSSION TITLE
Docs improvement: when to use river vs temporal

### DISCUSSION THREAD

**Discussion by JanRuettinger** [riverqueue/river#207] [Q&A]
I think it would be helpful to add a section to the docs on when to use River vs other popular services like Temporal.

My use case is the following:
1. User uploads 1-100 documents.
2. All documents need to get processed in three steps. All three steps depend on external services. Each step not take more than 5 seconds.
3. Inform user about progress updates, e.g. how many files were processed already & how many failed

Would it make more sense to use River for such a use case or is Temporal a better choice?

**Comment by bgentry** [comment]
River doesn't have anything built-in for this kind of thing today, although you could certainly build it yourself on top of the existing primitives. That being said we have some related stuff in the works that I think is going to make this kind of thing much easier in the near future 😎
